src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -108,6 +108,241 @@ public class AssertsFalseAttribute : Attribute
         public AssertsFalseAttribute () { }
     }
 }
+";
+
+        protected const string AsyncStreamsTypes = @"
+namespace System.Collections.Generic
+{
+    public interface IAsyncEnumerable<out T>
+    {
+        IAsyncEnumerator<T> GetAsyncEnumerator();
+    }
+
+    public interface IAsyncEnumerator<out T> : System.IAsyncDisposable
+    {
+        System.Threading.Tasks.ValueTask<bool> MoveNextAsync();
+        T Current { get; }
+    }
+}
+namespace System
+{
+    public interface IAsyncDisposable
+    {
+        System.Threading.Tasks.ValueTask DisposeAsync();
+    }
+}
+
+#nullable disable
```

Comment:
:memo: This code was moved from the async-streams tests, but I added a `#nullable disable` here to avoid nullable warnings in types that we only privately depend on.